### PR TITLE
Fix path_join() to not duplicate "/"

### PR DESCRIPTION
**Pull request by tristan957**
Off by one error...

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Review by nabeelmmd — APPROVED** [review]

File: lib/kvdb/kvdb_home.c [hunk 1]
```diff
@@ -42,7 +43,7 @@ path_join(const char *home, const char *path, char *buf, const size_t buf_sz)
         return 0;
     }
 
-    n = snprintf(buf, buf_sz, "%s%s%s", home, home[strlen(home)] == '/' ? "" : "/", path);
+    n = snprintf(buf, buf_sz, "%s%s%s", home, home[strlen(home) - 1] == '/' ? "" : "/", path);
```
Comment (beckerg) [inline lib/kvdb/kvdb_home.c:46]: Presumably strlen(home) cannot be zero...

Comment (tristan957) [reply lib/kvdb/kvdb_home.c:46]: Right. That's already been validated prior to this call

File: lib/mpool/src/mpool.c [hunk 2]
```diff
@@ -258,7 +258,8 @@ mpool_create(const char *home, const struct mpool_cparams *cparams)
             char   path[PATH_MAX];
             size_t n;
 
-            n = snprintf(path, sizeof(path), "%s/%s", home,
+            n = snprintf(path, sizeof(path), "%s%s%s", home,
+                         home[strlen(home) - 1] == '/' ? "" : "/",
```
Comment (beckerg) [inline lib/mpool/src/mpool.c:262]: Presumably you need at least one / at the end?  What if I give you a home with ten slashes at the end?

**Review by beckerg — APPROVED** [review]

File: lib/mpool/src/mpool.c (hunk 2 above)
Comment (tristan957) [reply lib/mpool/src/mpool.c:262]: No slash at the end is necessary. Regarding your other point, that could definitely be protected against. I can do that in another PR

**Pull request merged**